homeassistant/components/unifiprotect/media_player.py

```diff
@@ -112,10 +122,16 @@ async def async_play_media(
         self, media_type: str, media_id: str, **kwargs: Any
     ) -> None:
         """Play a piece of media."""
+        if media_source.is_media_source_id(media_id):
+            media_type = MEDIA_TYPE_MUSIC
+            play_item = await media_source.async_resolve_media(self.hass, media_id)
+            media_id = play_item.url
 
         if media_type != MEDIA_TYPE_MUSIC:
             raise ValueError("Only music media type is supported")
 
+        media_id = async_process_play_media_url(self.hass, media_id)
+
```

Comment:
The media is actually played by HA in a separate thread and streamed directly to the UniFi Protect camera using ffmpeg. The cameras cannot actually play media. They just have a UDP server running that you can send audio packets to.

Reply:
this turns relative URLs into full URLs and adds authentication. If it's not necessary, we can skip it.

Reply:
Yeah, it should not be necessary since ffmpeg is ran directly on HA.

Reply:
The relative URLs are still URLs, they are not local paths. Would that still be ok? 

Note that media sources _can_ resolve to a relative URL, but also to a full URL.

Reply:
Well, how to do differentate between a relative URL and a file path? If they both start with `/`, you would not be able to tell the difference. 

I guess you would `Path().exists()` it. If it is a file on disk pass it into `pyunifiprotect` directly, if it does not, convert it to a hass URL.

Reply:
that's not how media source works, it always resolves to a URL.

Reply:
But it may not always be a media source. If the user enters the value manually without the media browser it could be a file path.

Reply:
ok I will apply this only when resolving media sources.

If it can be a local path, the path needs to be checked that it's allowed to be streamed from using `hass.core.is_allowed_path`